pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java

```diff
@@ -328,6 +328,9 @@ Collection<PulsarSplit> getSplitsForTopic(String topicNamePersistenceEncoding,
                 splits.add(pulsarSplit);
             }
             return splits;
+        } catch (ManagedLedgerException exception){
```

Comment:
Do you want to catch the case that the managed ledger of the topic is not existing yet? Many exceptions extend `ManagedLedgerException`, maybe we don't need to catch all `ManagedLedgerException`.

Reply:
<img width="1431" alt="image" src="https://github.com/apache/pulsar/assets/15684156/de02d14f-8331-40ce-962b-9d706575af5d">


If we do not catch exceptions, the client will directly get this exception, but for the user, they do not need to know about this exception, I think we only need to return an empty list.


As for whether to catch all `ManagedLedgerException`, do you have any suggestions?

https://github.com/apache/pulsar/blob/2ab184e49a036a1dd10dc537bef4ab034a5ad5e0/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/AsyncCallbacks.java#L53

After all, the callback function declares `ManagedLedgerException` and cannot know which specific `ManagedLedgerException` will be thrown.

Reply:
Could you reproduce this problem? I don't know the root cause of the `NoSuchElementException`, it's hard to judge whether need to throw a server exception to clients, if this is a normal case, we don't need to notify clients, if this is abnormal, I think we need to send a server exception to clients, or else users will be confused by the empty result.

Reply:
Sorry, I don't know the root cause of the `NoSuchElementException`, it's hard to judge whether we need to throw a server exception to clients. If this is a normal case, we don't need to notify clients, if this is abnormal, I think we need to send a server exception to clients, or else users will be confused by the empty result, so we can only check the `ManagedLedgerException` reason is `NoSuchElementException` or not.

We'd better find the root cause of `NoSuchElementException` first, I think encountering `NoSuchElementException` while initializing ReadOnlyCursor is unreasonable.

Reply:
> If this is a normal case, we don't need to notify clients, if this is abnormal, I think we need to send a server exception to clients, or else users will be confused by the empty result.

I agree, I'll try to figure out the root cause of `NoSuchElementException` first.

Reply:
Having examined the source code, I know that when `ML` is initialized and sending messages, it will write data to `ledgers`. However, `ledgers` are always empty for a new topic, which results in a `NoSuchElementException`.


Therefore I think throwing this exception is normal, but for users using `pulsar-sql`, they should not get this exception. Returning an empty list to indicate that there is currently no data would be more reasonable.

```java
Caused by: java.util.NoSuchElementException
        at java.base/java.util.concurrent.ConcurrentSkipListMap.firstKey(ConcurrentSkipListMap.java:1859)
        at org.apache.bookkeeper.mledger.impl.ManagedLedgerImpl.getFirstPosition(ManagedLedgerImpl.java:3695)
        at org.apache.bookkeeper.mledger.impl.ReadOnlyCursorImpl.<init>(ReadOnlyCursorImpl.java:38)
        at org.apache.bookkeeper.mledger.impl.ReadOnlyManagedLedgerImpl.createReadOnlyCursor(ReadOnlyManagedLedgerImpl.java:138)
        at org.apache.bookkeeper.mledger.impl.ManagedLedgerFactoryImpl$4.openReadOnlyManagedLedgerComplete(ManagedLedgerFactoryImpl.java:498)
        at org.apache.bookkeeper.mledger.impl.ManagedLedgerFactoryImpl.lambda$asyncOpenReadOnlyManagedLedger$9(ManagedLedgerFactoryImpl.java:442)
        at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787)
        ... 16 more

```

Reply:
Thanks for your investigation, could we add a check in the method `ManagedLedgerImpl#getFirstPosition` to catch this case and return a null value?